check/src/kindcheck.rs

```diff
@@ -157,7 +157,7 @@ impl<'a> KindCheck<'a> {
                 gen.kind = try!(self.find(&gen.id));
                 Ok((gen.kind.clone(), Type::generic(gen)))
             }
-            Type::Variable(_) => panic!("kindcheck called on variable"),
+            Type::Variable(_) => Ok((self.subs.new_var(), typ.clone())),
```

Comment:
This was just a panic before since variables should never have been kindchecked on. Now however holes will have been replaced by variables before kindchecking so these must be of 'any' kind just as holes are.